streaming/src/main/scala/org/apache/spark/streaming/scheduler/BatchInfo.scala

```diff
@@ -36,8 +38,8 @@ case class BatchInfo(
     streamIdToInputInfo: Map[Int, StreamInputInfo],
     submissionTime: Long,
     processingStartTime: Option[Long],
-    processingEndTime: Option[Long]
-  ) {
+    processingEndTime: Option[Long],
+    failureReasons: Map[Int, String] = Map.empty) {
```

Comment:
I am not sure whether exposing the failureReason in this sort of a map is the right way. I am inclined to keep this private[streaming] in this PR (for both master and 1.5), so that we can atleast access it in the UI code. Later we can figure out whats the best way to expose this in master branch.